### PR TITLE
Include disabled bean info in beans endpoint

### PR DESCRIPTION
**Pull request by graemerocher**
- Refactor beans endpoint removing unnecessary reactive APIs
- Include disabled bean info in beans endpoint

### CODE REVIEW

**Comment by graemerocher** [comment]
Screenshot 2023-02-17 at 11 09 17

**Review by sdelamo — COMMENTED** [review]
Could you add a note into what's new? Telling that disabled beans are listed in the beans endpoint.

**Comment by graemerocher** [comment]
done.

File: inject/src/main/java/io/micronaut/context/DefaultBeanContext.java [hunk 1]
```diff
@@ -1629,6 +1629,14 @@ public Collection<BeanDefinition<?>> getAllBeanDefinitions() {
         return (Collection<BeanDefinition<?>>) Collections.emptyMap();
     }
 
+    @Override
+    public Collection<DisabledBean<?>> getDisabledBeans() {
+        return disabledBeans.values().stream()
+            .map(producer -> (DisabledBean<?>) producer.reference)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
```
Comment (alvarosanchez) [inline inject/src/main/java/io/micronaut/context/DefaultBeanContext.java:1637]: Java 17 `Stream::toList()`?

**Review by alvarosanchez — APPROVED** [review]

File: inject/src/main/java/io/micronaut/context/DefaultBeanContext.java (hunk 1 above)
Comment (graemerocher) [reply inject/src/main/java/io/micronaut/context/DefaultBeanContext.java:1637]: Tried, but doesn't compile

**Comment by graemerocher** [comment]
GraalVM failures are unrelated and fixed already

**Pull request merged**